File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
    local d = 0

    if a:len() ~= b:len() then
        return -1
    end

    for i = 1, a:len() do
        if a:sub(i, i) ~= b:sub(i, i) then
            d = d + 1
        end
    end

    return d
end

return Hamming

File: go/hamming/hamming.go
package hamming

import "fmt"

// Distance will compute hamming distance between to DNA strands
func Distance(a, b string) (int, error) {
	var diff int

	if len(a) != len(b) {
		return -1, fmt.Errorf("DNA strands lengths not equals")
	}

	for i := range a {
		if a[i] != b[i] {
			diff++
		}
	}

	return diff, nil
}
